File: 박소현/2기/2주차/짝지어 제거하기.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(String s) {
        String[] str = s.split("");
        Stack<String> st = new Stack<>();
        st.add(str[0]); // 맨 처음 문자는 스택에 넣고 시작하자
        
        // 스택이 비어있지 않다면 peek한 것과 비교했을 때 같을 경우 pop하기
        // 그렇지 않다면 무조건 add하기
        for(int i=1; i<str.length; i++) {
            if (!st.isEmpty() && st.peek().equals(str[i])) st.pop();
            else st.add(str[i]);
        }
        
        int answer = -1;
        answer= st.isEmpty()? 1:0;

        return answer;
    }
}

File: 이준혁/2기/6주차/짝지어제거하기.js
function solution(s)
{
    // 입력이 홀수라면 2개씩 나눌 수 없다.
    if (s.length%2 === 1) {
        return 0; // 그러므로 0 리턴
    }

    let stack = []; // 스택 사용

    for (let i=0; i<s.length; i++) { 
        // stack에 늦게 들어온 값이 현재 s[i]와 같다면
        if (stack[stack.length-1] === s[i]) {
            stack.pop(); // 짝이 맞으므로 지움 
        } else {
            stack.push(s[i]); // 아니라면 저장
        }
    }

    if (stack.length === 0) {
        return 1;
    } else {
        return 0;
    }
}
